seriate.py

```diff
@@ -11,9 +11,47 @@
 ortools_version = Version(ortools.__version__)
 ortools6 = Version("6.0.0") <= ortools_version < Version("7")
 ortools7 = Version("7.0.0") <= ortools_version < Version("8")
+if not ortools6 and not ortools7:
+    raise ImportError("No valid version of ortools installed. Please install ortools 6 or 7.")
 
 
-def seriate(dists: numpy.ndarray, approximation_multiplier=1000, timeout=2.0) -> List[int]:
+class IncompleteSolutionError(Exception):
+    pass
+
+
+def seriate(dists: numpy.ndarray, approximation_multiplier: int = 1000,
+            timeout: float = 2.0) -> List[int]:
+    """
+    Order the elements of a set so that the sum of sequential pairwise distances is minimal.
+
+    We solve the Travelling Salesman Problem (TSP) under the hood.
+    Reference: http://nicolas.kruchten.com/content/2018/02/seriation/
+
+    :param dists: Either a condensed pdist-like or a symmetric square distance matrix.
+    :param approximation_multiplier: Multiply by this number before converting distances \
+                                     to integers.
+    :param timeout: Maximum amount of time allowed to spend for solving the TSP, in seconds. \
+                    If timeout is 0 it will try to solve the problem with timeout = 1, and \
+                    double the timeout every time it fails until a valid solution is found.
+    :return: List with ordered element indexes, the same length as the number of elements \
+             involved in calculating `dists`.
+    """
+    if timeout > 0:
+        return _seriate(dists=dists, approximation_multiplier=approximation_multiplier,
+                        timeout=timeout)
+    timeout = 1.
+    route = None
+    while route is None:
+        try:
+            route = _seriate(dists=dists, approximation_multiplier=approximation_multiplier,
+                             timeout=timeout)
+            break
+        except IncompleteSolutionError:
+            timeout *= 2
+    return route
+
+
+def _seriate(dists: numpy.ndarray, approximation_multiplier=1000, timeout=2.0) -> List[int]:
```

Comment:
Could you please change the docstring to indicate that the timeout is always greater than 0. Also `assert` on it.

Reply:
@Guillemdb This still holds.

Reply:
`assert` should never be used to validate use input, as it meant to be used as a debugging tool that can be disabled in production. A ValueError should be raised instead to be sure that even disabling the asserts it cannot run with a timeout less than zero.